### PR TITLE
added push notification for when user leaves app in bg

### PR DESCRIPTION
**Pull request by timberwang0810**
Screen Shot 2021-11-29 at 3 39 11 PM

### CODE REVIEW

File: StudyBuddyApp/StudyBuddyApp/Views/DoingTaskView.swift [hunk 1]
```diff
@@ -113,6 +122,27 @@ struct DoingTaskView: View {
           }
         }
       }
+    }.onReceive(NotificationCenter.default.publisher(for: UIApplication.didEnterBackgroundNotification)) { _ in
+      print("Moving to the background!")
+      self.pauseTimer()
+      let content = UNMutableNotificationContent()
+      content.title = "Come back!🥺"
```
Comment (austin-leung) [inline StudyBuddyApp/StudyBuddyApp/Views/DoingTaskView.swift:129]: ...

**Review by austin-leung — APPROVED** [review]
Thanks for fixing the bug with items not showing up immediately on playground view.

I'm assuming we have no fix for the termination of app for now which is ok?

Also one small piece of behavior I noticed is that if you pause the task and then leave app, clicking notification back into app automatically plays the timer. Not sure if that's expected behavior but it's not the biggest issue. Maybe better behavior would be to always have the task paused when you re-enter the app.

File: StudyBuddyApp/StudyBuddyApp/Views/DoingTaskView.swift (hunk 1 above)
Comment (timberwang0810) [reply StudyBuddyApp/StudyBuddyApp/Views/DoingTaskView.swift:129]: xd

**Comment by timberwang0810** [comment]
> Thanks for fixing the bug with items not showing up immediately on playground view.
> 
> I'm assuming we have no fix for the termination of app for now which is ok?
> 
> Also one small piece of behavior I noticed is that if you pause the task and then leave app, clicking notification back into app automatically plays the timer. Not sure if that's expected behavior but it's not the biggest issue. Maybe better behavior would be to always have the task paused when you re-enter the app.

yep i will look at termination stuff next week perhaps. And yes that's the behavior I coded in, and I think if we want it the other way it's just a matter of commenting out a line

**Pull request merged**